integration_test/unreadmarker_test.dart

```diff
@@ -0,0 +1,65 @@
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+import 'package:zulip/widgets/message_list.dart';
+import 'package:zulip/widgets/store.dart';
+
+import '../test/api/fake_api.dart';
+import '../test/example_data.dart' as eg;
+import '../test/model/binding.dart';
+import '../test/model/message_list_test.dart';
+
+void main() {
+  final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  TestZulipBinding.ensureInitialized();
+
+  late PerAccountStore store;
+  late FakeApiConnection connection;
+
+  Future<List<Message>> setupMessageListPage(WidgetTester tester, int messageCount) async {
+    addTearDown(testBinding.reset);
+    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+    store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+    connection = store.connection as FakeApiConnection;
+
+    // prepare message list data
+    final messages = List.generate(messageCount,
+      (i) => eg.streamMessage(flags: [MessageFlag.read]));
+    connection.prepare(json:
+      newestResult(foundOldest: true, messages: messages).toJson());
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: GlobalStoreWidget(
+          child: PerAccountStoreWidget(
+            accountId: eg.selfAccount.id,
+            child: const MessageListPage(narrow: AllMessagesNarrow())))));
+    await tester.pumpAndSettle();
+    return messages;
+  }
+
+  testWidgets('_UnreadMarker animation performance test', (tester) async {
+    final messages = await setupMessageListPage(tester, 500);
+
+    // short pause for buffer during video-capture
+    await tester.pump(const Duration(seconds: 2));
```

Comment:
Interesting — what's the video-capture about? Is that involved in the performance test?

Reply:
Removing this for now. I also used this integration test to automate the app for video recording purposes, and it was useful to have a bit of buffer before the action started, but that isn't relevant to a performance test.